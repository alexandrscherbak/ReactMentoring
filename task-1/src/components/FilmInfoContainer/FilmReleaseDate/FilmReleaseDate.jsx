import React from 'react';

const FilmReleaseDate = ({releaseDate}) => (
	<div>
		<span>{releaseDate.slice(0, 4)}</span>

		<style jsx>
			{`
				span {
					color: #9e9e9e;
					border: 1px solid #9e9e9e;
				}
			`}
		</style>
	</div>
);

export default FilmReleaseDate;
