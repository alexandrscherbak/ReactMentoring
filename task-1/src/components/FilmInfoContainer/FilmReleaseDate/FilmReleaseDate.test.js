import React from 'react';
import Enzyme, { mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import FilmReleaseDate from './FilmReleaseDate.jsx';
import renderer from 'react-test-renderer';

Enzyme.configure({ adapter: new Adapter() });

const props = {
	releaseDate: '2018-04-01',
};

function setup() {
	return mount(<FilmReleaseDate {...props}/>);
}

describe('FilmReleaseDate', () => {
	test('snapshot', () => {
		const component = renderer.create(
			<FilmReleaseDate {...props}/>,
		);
		let tree = component.toJSON();
		expect(tree).toMatchSnapshot();
	});
	it('should render self and subcomponents', () => {
		const enzymeWrapper = setup();
		expect(enzymeWrapper.find('span').text()).toBe(props.releaseDate.slice(0, 4));
	});
});
