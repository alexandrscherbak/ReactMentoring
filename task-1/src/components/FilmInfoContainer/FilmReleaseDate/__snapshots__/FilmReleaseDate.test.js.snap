// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilmReleaseDate snapshot 1`] = `
<div
  className="jsx-617062702"
>
  <span
    className="jsx-617062702"
  >
    2018
  </span>
</div>
`;
